09_json/willshen8/cmd/puppy-server/main.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/anz-bank/go-course/09_json/willshen8/pkg/puppy"
+	jsonify "github.com/anz-bank/go-course/09_json/willshen8/pkg/puppy/json"
+	store "github.com/anz-bank/go-course/09_json/willshen8/pkg/puppy/store"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var out io.Writer = os.Stdout
+
+var (
+	args     = os.Args[1:]
+	fileName = kingpin.Flag("data", "Puppy Json File Name").Short('d').String()
```

Comment:
maybe better use [`File()`](https://godoc.org/gopkg.in/alecthomas/kingpin.v2#ArgClause.File) - does the error handling for you if file does not exist.

Reply:
Hi @juliaogris , I kind of disagree with using `File()` in this case. As the two possible file reading operations: `os.Open` and `ioutil.ReadFile` takes a string as an argument. `ioutil.ReadFile` is more appropriate for this instance as we just get a blob of data, and have no need to read data sequentially. Having a file pointer returned from `File()` just adds extra work. I can certainly add more checks on whether a file exists or whether we have permission to the file, etc. But I rely on your expertise to more guidance.

Reply:
You certainly get a warning/panic if the file doesn't exist, for example:
```
go run main.go --data fdsfsdfsdf.txt
panic: open fdsfsdfsdf.txt: no such file or directory
```

Reply:
Thanks, I have used `ExistingFile()` which is perhaps more correct grammarly.